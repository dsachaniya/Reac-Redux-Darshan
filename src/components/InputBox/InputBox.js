import React from 'react';
import Proptypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

/**
 * Home
 * @param {Object} props
 * @returns {Node}
 */
const InputBox = (props) => {
  const { type, onChange, helperText, autoFocus, onBlur, id } = props;
  return (
    <section>
      <TextField
        variant="outlined"
        margin="normal"
        required
        fullWidth
        label={type}
        name={type}
        autoComplete={type}
        type={type}
        autoFocus={autoFocus}
        onChange={onChange}
        helperText={helperText}
        error={!!helperText}
        onBlur={onBlur}
        id={id}
      />
    </section>
  );
};

InputBox.propTypes = {
  type: Proptypes.string.isRequired,
  onChange: Proptypes.func.isRequired,
  helperText: Proptypes.string,
  autoFocus: Proptypes.bool,
  onBlur: Proptypes.func.isRequired,
  id: Proptypes.string.isRequired,
};

InputBox.defaultProps = {
  autoFocus: false,
  helperText: '',
};

export default InputBox;
