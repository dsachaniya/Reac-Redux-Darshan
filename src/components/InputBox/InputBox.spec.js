import React from 'react';
import { shallow } from 'enzyme';
import InputBox from './InputBox';

describe('<InputBox />', () => {
  it('Rendering', () => {
    const wrapper = shallow(<InputBox />);
    expect(wrapper).toMatchSnapshot();
  });
});
