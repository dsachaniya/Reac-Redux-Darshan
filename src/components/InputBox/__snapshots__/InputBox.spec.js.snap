// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputBox /> Rendering 1`] = `
<section>
  <WithStyles(ForwardRef(TextField))
    autoFocus={false}
    error={false}
    fullWidth={true}
    helperText=""
    margin="normal"
    required={true}
    variant="outlined"
  />
</section>
`;
